### PR TITLE
Fleshes out settings screen with groups, items, and navigation

### PR DESCRIPTION
**Pull request by RedBrogdon**
Adds refactored versions of SettingsItem and SettingsGroup from @xster's demo, as well as a functional main settings screen and two routes for settings details.

### CODE REVIEW

**Comment by xster** [comment]
Add a code comment note in the table view related files that they're going to be replaced by actual widgets in Cupertino once https://github.com/flutter/flutter/projects/29 is done.

**Comment by RedBrogdon** [comment]
> Add a code comment note in the table view related files that they're going to be replaced by actual widgets in Cupertino once https://github.com/flutter/flutter/projects/29 is done.

No problem. Which files do you mean, just so I can get them all in one go? Anything that's presenting a large, scrolling list of items like the home, search, and favorite screens?

**Comment by xster** [comment]
settings_group and settings_item

File: veggieseasons/lib/screens/settings.dart [hunk 1]
```diff
@@ -3,19 +3,199 @@
...
+class VeggieCategorySettingsScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final model = ScopedModel.of<Preferences>(context, rebuildOnChange: true);
+    final currentPrefs = model.preferredCategories;
+
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text('Preferred Categories'),
+        previousPageTitle: 'Settings',
```
Comment (xster) [inline veggieseasons/lib/screens/settings.dart]: You might not need this if your previous CupertinoPageRoute has a `title` field.

File: veggieseasons/lib/screens/settings.dart [hunk 2]
```diff
@@ -3,19 +3,199 @@
...
+                    model.addPreferredCategory(category);
+                  } else {
+                    model.removePreferredCategory(category);
+                  }
+                },
+              );
+            } else {
+              toggle = CupertinoSwitch(
+                value: false,
+                onChanged: (value) {},
```
Comment (xster) [inline veggieseasons/lib/screens/settings.dart]: This is not disabled though right?

File: veggieseasons/lib/screens/settings.dart [hunk 3]
```diff
@@ -3,19 +3,199 @@
...
 
-class SettingsScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
+    final model = ScopedModel.of<Preferences>(context, rebuildOnChange: true);
+
     return CupertinoPageScaffold(
       navigationBar: CupertinoNavigationBar(
-        middle: Text('Settings'),
+        middle: Text('Target Calorie Level'),
```
Comment (xster) [inline veggieseasons/lib/screens/settings.dart]: Seems long (e.g. < Settings   Target Calorie Level). Are you sure this fits on an iPhone SE?

File: veggieseasons/lib/screens/settings.dart [hunk 4]
```diff
@@ -3,19 +3,199 @@
...
+    return SettingsItem(
+      label: 'Target Calorie Intake',
+      icon: SettingsIcon(
+        backgroundColor: Styles.iconBlue,
+        icon: Styles.calorieIcon,
+      ),
+      content: FutureBuilder<int>(
+        future: prefs.desiredCalories,
+        builder: (context, snapshot) => Text(snapshot.data?.toString() ?? ''),
+      ),
```
Comment (xster) [inline veggieseasons/lib/screens/settings.dart]: I may not visualize correctly but all table cell items that navigates needs the > chevron at the end

File: veggieseasons/lib/screens/settings.dart (hunk 1 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/settings.dart]: Coming from Material-based stuff, I didn't realize that property existed. Updated.

File: veggieseasons/lib/screens/settings.dart (hunk 3 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/settings.dart]: It fits on an SE, but looks crowded. Revised to be 'Calorie Target' here and elsewhere.

File: veggieseasons/lib/screens/settings.dart (hunk 2 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/settings.dart]: No. It should be `onChanged: null`.

Out of curiosity, if null is an acceptable value, why are the onChanged callbacks for switches `@required` (both here and in material.dart)?

File: veggieseasons/lib/screens/settings.dart (hunk 4 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/settings.dart]: Added. Should I pull the preview of the selected value as well, or include both?

**Comment by RedBrogdon** [comment]
Comments addressed, also added disclaimers to the settings_group and settings_item files. PTAL!

File: veggieseasons/lib/screens/settings.dart (hunk 2 above)
Comment (xster) [reply veggieseasons/lib/screens/settings.dart]: Just to make it explicit

File: veggieseasons/lib/screens/settings.dart (hunk 4 above)
Comment (xster) [reply veggieseasons/lib/screens/settings.dart]: You can have both

**Comment by xster** [comment]
LGTM

**Pull request merged**